tests/parser/ParseContextTests.cpp

```diff
@@ -748,3 +748,25 @@ BOOST_AUTO_TEST_CASE(Test_EXIT1) {
   the extra process - that kind of feelt a bit over the top.
 */
 }
+
+
+/*
+  If there are errors the ErrorGuard destructor will print error messages and
+  call std::terminate() in the test we therefor create the instance with errors
+  using new - and let it leak.
```

Comment:
I really do not like that...! Same would go for use in Flow, I would really not be happy about terminate() being a possible (not really that exceptional) possibility, would rather see an exception used.

Reply:
I will try again on the exception path.